### PR TITLE
Enable completion-only loss in SFTTrainer when using Liger Kernel

### LINKED ISSUE #3484: Completions Only Loss is incompatible with use_liger_kernel set as true
Hi Team, discovered that completions only loss does not work when use_liger_kernel is set as true, since this line drops the completion_mask. Is this an expected behavior? 

https://github.com/huggingface/trl/blob/65245f6be8dea3d559a58788eca439f9c2d0f1e2/trl/trainer/sft_trainer.py#L670

### PR DESCRIPTION
**Pull request by kswhitecross**
# What does this PR do?

Fixes #3484, where the `'completion_mask'` column will be stripped from `trainer.train_dataset` and `trainer.eval_dataset` if `args.use_liger_kernel == True`.  This issue makes it impossible to use the SFTTrainer to train on prompt-completion style datasets with the memory-efficient Liger kernel, without preprocessing the dataset manually. 

This issue can be minimally reproduced with the following example:

```python
from trl import SFTTrainer, SFTConfig
from datasets import Dataset

train_dataset = Dataset.from_dict({
    "prompt": ["What is the capital of France?", "What is the capital of Germany?"],
    "completion": [" The capital of France is Paris.", " The capital of Germany is Berlin."]
})

training_args = SFTConfig(
    use_liger_kernel=True,
    completion_only_loss=True
)

trainer = SFTTrainer(
    model="Qwen/Qwen2.5-0.5B",
    train_dataset=train_dataset,
    args=training_args
)

print(trainer.train_dataset)
```
which prints:
```
Dataset({
    features: ['input_ids'],  # this is missing 'completion_mask', so loss will be computed on the prompt as well!
    num_rows: 2
})
```

With this PR, the output will instead be
```
Dataset({
    features: ['completion_mask', 'input_ids'],
    num_rows: 2
})
```
enabling the masking out of prompt tokens from the loss.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ x ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ x ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by kswhitecross** [comment]
Thanks @shirinyamani !  This is my first PR

I tried to run `make precommit` on my own machine, but got an error including `make: pre-commit: No such file or directory`.  Is `pre-commit` missing from the `[dev]` requirements?

**Comment by kashif** [comment]
@kswhitecross  yes do `pip install precommit`

**Comment by shirinyamani** [comment]
> Thanks @shirinyamani ! This is my first PR
> 
> I tried to run `make precommit` on my own machine, but got an error including `make: pre-commit: No such file or directory`. Is `pre-commit` missing from the `[dev]` requirements?


Have you installed `pip install precommit` and `ruff`?
this happened to me before, actually there are two precommits when you run the command is `make precommit` without **-** in between but when it actually checks for formatting or style stuff it requires you to `pip install pre-commit`, im personally not 100% sure why this happens but reading the make file and installing what i mentioned above helped me figuring the issue in past!

**Comment by shirinyamani** [comment]
@kswhitecross but anyways, thanks for the catch and  for now i ran it and CI failure is not relevant to this PR.  So we are good on this PR for now!

**Comment by sonalexle** [comment]
Could you please do the same for assistant_masks? Thanks!
